transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CCTDestination.java

```diff
@@ -17,22 +17,137 @@
 
 public final class CCTDestination implements Destination {
   static final String DESTINATION_NAME = "cct";
-  public static final CCTDestination INSTANCE = new CCTDestination();
+  static final String LEGACY_DESTINATION_NAME = "lflg";
 
-  private CCTDestination() {}
+  private static final String DEFAULT_API_KEY =
+      StringMerger.mergeStrings("AzSCki82AwsLzKd5O8zo", "IayckHiZRO1EFl1aGoK");
+  private static final String EXTRAS_VERSION_MARKER = "1$";
+  private static final String EXTRAS_DELIMITER = "\\";
+
+  public static final CCTDestination DEFAULT_INSTANCE =
+      new CCTDestination(DESTINATION_NAME, null, null);
+  public static final CCTDestination DEFAULT_LEGACY_INSTANCE =
+      new CCTDestination(LEGACY_DESTINATION_NAME, DEFAULT_API_KEY, null);
+
+  private final String destinationName;
+  private final String apiKey;
+  private final String endPoint;
+
+  private CCTDestination(
+      @NonNull String destinationName, @Nullable String apiKey, @Nullable String endPoint) {
+    this.destinationName = destinationName;
+    this.apiKey = apiKey;
+    this.endPoint = endPoint;
+  }
+
+  @NonNull
+  public static CCTDestination customLegacyDestination(
+      @Nullable String apiKey, @Nullable String endPoint) {
+    return new CCTDestination(LEGACY_DESTINATION_NAME, apiKey, endPoint);
+  }
 
   @NonNull
   @Override
   public String getName() {
-    return DESTINATION_NAME;
+    return destinationName;
   }
 
   @Nullable
   @Override
   public byte[] getExtras() {
-    return null;
+    return asByteArray();
+  }
+
+  @Nullable
+  public String getAPIKey() {
+    return apiKey;
+  }
+
+  @Nullable
+  public String getEndPoint() {
+    return endPoint;
+  }
+
+  /**
+   * Encodes the current object as a byte[].
+   *
+   * <p><strong>Important:</strong> Changes in the encoding should be backwards compatible to ensure
+   * any data persisted from a previous version can be read successfully.
+   */
+  @Nullable
+  public byte[] asByteArray() {
+    if (apiKey == null && endPoint == null) {
+      return null;
+    }
+    String buffer =
+        String.format(
+            "%s%s%s%s%s%s",
+            EXTRAS_VERSION_MARKER,
+            destinationName,
+            EXTRAS_DELIMITER,
+            endPoint == null ? "" : endPoint,
+            EXTRAS_DELIMITER,
+            apiKey == null ? "" : apiKey);
+    try {
+      return buffer.getBytes("UTF-8");
+    } catch (UnsupportedEncodingException e) {
+      throw new IllegalStateException("UTF-8 encoding not found.");
+    }
+  }
+
+  /**
+   * Creates a new object by parsing a {@code byte[]} generated by {@link asByteArray}.
+   *
+   * <p><strong>Important:</strong> Changes in the encoding should be backwards compatible to ensure
+   * any data persisted from a previous version can be read successfully.
+   */
+  @NonNull
+  public static CCTDestination fromByteArray(@NonNull byte[] a) {
+    String buffer;
+    try {
+      buffer = new String(a, "UTF-8");
```

Comment:
I think if you just use `Charset.forName("UTF-8")` you won't have to deal with checked exceptions. Here and throughout.